### PR TITLE
Add Rich-Content Feature to Landing Page

### PR DESCRIPTION
**Pull request by it-tma-tri**
## Summary by Sourcery

This pull request introduces rich-text content support to the landing page, refactors several components for better code quality, and enhances the handling of user mentions and media sources.

- **New Features**:
    - Added rich-text content support to the landing page, allowing for more dynamic and formatted content display.
- **Enhancements**:
    - Refactored the ChatContent component to improve readability and maintainability.
    - Updated the LandingPage component to include the new RichTextContent component and adjusted the layout accordingly.
    - Enhanced the MediaSource component by cleaning up the code and improving the handling of YouTube URLs and language changes.
    - Improved the ChatInput component by adding a watcher for the autocomplete search to handle user mentions more effectively.

### CODE REVIEW

**Comment by techyay** [comment]
Please fix indentation and add validations.

File: webapp/src/views/admin/rooms/types-edit/page-landing.vue [hunk 1]
```diff
@@ -4,13 +4,15 @@
 	upload-url-input(v-model="modules['page.landing'].config.header_background_image", label="Header background image", name="header-background-image")
 	bunt-input(v-model="modules['page.landing'].config.header_background_color", label="Header background color", name="headerbackgroundcolor")
 	bunt-select(v-model="modules['page.landing'].config.sponsor_room_id", label="Sponsor Source Room", name="sponsor-room", :options="sponsorRooms", option-label="name")
+	rich-text-editor(label="Main content", v-model="modules['page.landing'].config.main_content")
```
[the remark this answers is not in the document]
Comment (hongquan) [reply webapp/src/views/admin/rooms/types-edit/page-landing.vue:7]: Will do later.

**Review by hongquan — APPROVED** [review]

**Comment by hongquan** [comment]
Please don't add "fix-141:" to the title (make the title unnecessary long).
Specifying it in the description is enough.

**Comment by mariobehling** [comment]
@sourcery-ai review

File: webapp/src/components/ChatContent.vue [hunk 2]
```diff
@@ -40,55 +40,16 @@ export async function contentToPlainText (content) {
...
-      }
-      return createElement('span', { domProps: { innerHTML: part.html } })
-    })
-  },
-  components: {
-    'user-modal': {
-      template: `
+	functional: true,
+	components: {
+		props: ['selectedUser'],
```
[the remark this answers is not in the document]
Comment (odkhang) [reply webapp/src/components/ChatContent.vue:50]: image

moving props out will cause lint error

**Pull request merged**